VisualizationBase/src/items/ViewItem.cpp

```diff
@@ -142,6 +130,63 @@ Model::Node* ViewItem::nodeAt(int column, int row) const
 	return nodes_[column][row];
 }
 
+void ViewItem::notifyAboutRemoval(Item *item)
```

Comment:
I'm not happy about the names of this and the next method. `notify` typically is used in contexts where the signals and slots mechanism is used, which is not the case here. Could you please rename them to:
* `cleanupRemovedNode`
* `cleanupRemovedItem`

or some other more appropriate name.